gce/startup-script.sh

```diff
@@ -0,0 +1,52 @@
+# Copyright 2019 Google LLC All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Echo commands
+set -v
+
+# Variables for finding source code
+PROJECTID=$(curl -s "http://metadata.google.internal/computeMetadata/v1/project/project-id" -H "Metadata-Flavor: Google")
+REPOSITORY=[YOUR_REPO_NAME]
+
+# Install Stackdriver logging agent
+curl -sSO https://dl.google.com/cloudagents/install-logging-agent.sh
+sudo bash install-logging-agent.sh
+
+# Install or update needed software
+apt-get update
+apt-get install -yq git supervisor python python-pip
+pip install --upgrade pip virtualenv
+
+# Account to own server process
+useradd -m -d /home/pythonapp pythonapp
+
+# Fetch source code
+export HOME=/root
+git config --global credential.helper gcloud.sh
+git clone https://source.developers.google.com/p/$PROJECTID/r/$REPOSITORY /opt/app
```

Comment:
For PHP I am removing the integration with GCR and instead cloning the app straight from github and installing the dependencies on the machine.

This is because, with the simplified app it's no longer necessary to make changes to a file and deploy to a new repo. And I do not think that deploying from GCR is canonical enough of a solution to merit inclusion in a hello world tutorial.

Please let me know if you feel otherwise! If not, these lines can be replaced with cloning from github.

Reply:
I feel the same. I'll be happy to just get from the public GitHub directory.